spec/Clay/MediaSpec.hs

```diff
@@ -0,0 +1,98 @@
+    "min-device-width: 18em" `shouldRenderFromFeature` minDeviceWidth (em 18)
+    "max-device-width: 19em" `shouldRenderFromFeature` maxDeviceWidth (em 19)
+    "device-height: 20em" `shouldRenderFromFeature` deviceHeight (em 20)
+    "min-device-height: 21em" `shouldRenderFromFeature` minDeviceHeight (em 21)
+    "max-device-height: 22em" `shouldRenderFromFeature` maxDeviceHeight (em 22)
+
+  describe "aspect ratio features" $ do
+    "aspect-ratio: 2/3" `shouldRenderFromFeature` aspectRatio (2, 3)
+    "min-aspect-ratio: 4/5" `shouldRenderFromFeature` minAspectRatio (4, 5)
+    "max-aspect-ratio: 6/7" `shouldRenderFromFeature` maxAspectRatio (6, 7)
+    "device-aspect-ratio: 2/3" `shouldRenderFromFeature` deviceAspectRatio (2, 3)
+    "min-device-aspect-ratio: 4/5" `shouldRenderFromFeature` minDeviceAspectRatio (4, 5)
+    "max-device-aspect-ratio: 6/7" `shouldRenderFromFeature` maxDeviceAspectRatio (6, 7)
+
+  describe "color features" $ do
+    "color" `shouldRenderFromFeature` color
+    "monochrome" `shouldRenderFromFeature` monochrome
+    "scan" `shouldRenderFromFeature` scan
+    "grid" `shouldRenderFromFeature` grid
+    "min-color: 23" `shouldRenderFromFeature` minColor 23
+    "max-color: 25" `shouldRenderFromFeature` maxColor 25
+    "color-index: 15" `shouldRenderFromFeature` colorIndex 15
+    "min-color-index: 17" `shouldRenderFromFeature` minColorIndex 17
+    "max-color-index: 19" `shouldRenderFromFeature` maxColorIndex 19
+    "min-monochrome: 77" `shouldRenderFromFeature` minMonochrome 77
+    "max-monochrome: 99" `shouldRenderFromFeature` maxMonochrome 99
+
+  describe "resolution features" $ do
+    "resolution: 45dpi" `shouldRenderFromFeature` resolution (dpi 45)
+    "resolution: 74dppx" `shouldRenderFromFeature` resolution (dppx 74)
+    "min-resolution: 45dpi" `shouldRenderFromFeature` minResolution (dpi 45)
+    "min-resolution: 74dppx" `shouldRenderFromFeature` minResolution (dppx 74)
+    "max-resolution: 45dpi" `shouldRenderFromFeature` maxResolution (dpi 45)
+    "max-resolution: 74dppx" `shouldRenderFromFeature` maxResolution (dppx 74)
+
+  describe "preference features" $ do
+    "prefers-color-scheme: light" `shouldRenderFromFeature` prefersColorScheme light
+    "prefers-color-scheme: dark" `shouldRenderFromFeature` prefersColorScheme dark
+
+-- | Empty CSS for when CSS is needed but we don't care about it.
+emptyStyle :: Css
+emptyStyle = pure ()
+
+-- | The text should be rendered from the media type.
+shouldRenderFromMediaType :: Text -> MediaType -> SpecWith ()
+shouldRenderFromMediaType text mediaType =
+  shouldRenderAsFrom (unpack text) fullText css
+  where
+    fullText = "@media " <> text <> "{}"
+    css = query mediaType [] emptyStyle
+
+-- | The text should be rendered from the feature.
+shouldRenderFromFeature :: Text -> Feature -> SpecWith ()
+shouldRenderFromFeature text feature =
+  shouldRenderAsFrom (unpack text) fullText css
+  where
+    fullText = "@media all and (" <> text <> "){}"
```

Comment:
Should a feature always render with brackets?

Reply:
According to https://w3c.github.io/csswg-drafts/mediaqueries-5/#mq-syntax, the parentheses are mandatory.  One can note that the `<mf-boolean>` syntax element, of which `color`, `monochrome`, etc. correspond to, is always required to be surrounded by parentheses in the `<media-feature>` element.

(When the tests for `color`, `monochrome`, etc. failed, I tried to find references to see if the parentheses were mandatory were not, i.e., whether I had to change the tests.  I had to go to an official specification because most references and guides only had examples with them in parentheses without indicating whether they were mandatory or not.)